### PR TITLE
fix: role=radio should not require aria-checked

### PR DESCRIPTION
**Pull request by jkodu**
As per ARIA spec, when an element has `role=radio`, states and properties are derived from the role and do not have be explicitly mentioned.

See - https://www.w3.org/TR/wai-aria-1.1/#radio for more info.

Closes issue:
- NA

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [x] Follows the commit message policy, appropriate for next version
- [x] Has documentation updated, a DU ticket, or requires no documentation change
- [x] Includes new tests, or was unnecessary
- [x] Code is reviewed for security by: @WilcoFiers

### CODE REVIEW

File: test/integration/rules/aria-required-attr/required-attr.html [hunk 1]
```diff
@@ -3,6 +3,7 @@
 <div role="spinbutton" id="pass3" aria-valuenow="value" aria-valuemax="value" aria-valuemin="value">ok</div>
 <div role="heading" id="pass4" aria-level="1">ok</div>
 <div role="combobox" id="pass5" aria-expanded="true">ok</div>
+<span role="radio" id="pass6">I am BLUE!</span>
```
Comment (WilcoFiers) [inline test/integration/rules/aria-required-attr/required-attr.html]: Add a check to show it doesn't fail when you give it `aria-checked` please.

Comment (WilcoFiers) [reply test/integration/rules/aria-required-attr/required-attr.html]: ... in the rule that checks allowed attributes.

**Comment by jkodu** [comment]
@WilcoFiers - wai tools need this, so a review/ approval can be handy.

**Comment by aweary** [comment]
> As per ARIA spec, when an element has role=radio, states and properties are derived from the role and do not have be explicitly mentioned.

If I'm looking at [the spec](https://www.w3.org/TR/wai-aria-1.1/#radio) correctly, `aria-checked` is marked as a required property, not inherited. The spec says that this identifies:

> States and properties specifically required for the role and subclass roles. Content authors MUST provide a non-empty value for required states and properties.

@WilcoFiers @jkodu it seems like this should still be marked as required?

**Pull request merged**